src/pages/index.js

```diff
@@ -3,16 +3,139 @@ import FormValidator from "../components/FormValidator.js";
 import "./index.css";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import confirmPopup from "../components/confirmPopup.js";
+import Popup from "../components/Popup.js";
 import Section from "../components/Section";
 import UserInfo from "../components/UserInfo.js";
-import { data, variables, formConfig } from "../utils/Constants.js";
+import { variables, formConfig } from "../utils/Constants.js";
+import Api from "../components/Api.js";
 
 //------------------------------------------------ Refactoring Code
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    Authorization: "44ea7345-0378-4698-8841-25c6e07018e7",
+    "Content-Type": "application/json",
+  },
+});
+
+const user = new UserInfo({
+  nameSelector: "#profile-info-title",
+  jobSelector: "#profile-info-description",
+  avatarSelector: "#user-avatar",
+});
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    user.setUserInfo({
+      name: userData.name,
+      job: userData.about,
+      avatar: userData.avatar,
+    });
+    console.log("User information loaded and displayed successfully.");
+  })
+  .catch((error) => {
+    console.error("Failed to load user information:", error);
+  });
+
+const editFormPopup = new PopupWithForm("#profile-edit-modal", (formData) => {
+  variables.editModalProfileSaveButton.textContent = "Saving...";
+  api
+    .updateUserProfile({ name: formData.title, about: formData.description })
+    .then(() => {
+      variables.editModalProfileSaveButton.textContent = "Save";
+      user.setUserInfo({
+        name: formData.title,
+        job: formData.description,
+      });
+    });
+});
 
-function createCard(item) {
-  const card = new Card(item, "#elementCard", handleImageClick);
-  return card.getView();
+variables.profileEditButton.addEventListener("click", () => {
+  const userInfo = user.getUserInfo();
+  variables.profileModalNameInput.value = userInfo.name.trim();
+  variables.profileModalDescriptionInput.value = userInfo.job.trim();
+  editFormPopup.open();
+});
+
+editFormPopup.setEventListeners();
+
+getCards();
+
+function getCards() {
+  api
+    .getInitialCards()
+    .then((cards) => {
+      console.log("Cards fetched and rendered successfully.");
+      renderCards(cards);
+      return cards;
+    })
+    .catch((error) => {
+      console.error("Failed to fetch cards:", error);
+    });
 }
+
+function renderCards(cards) {
+  cards.forEach((card) => {
+    addCardToCardSection(card);
```

Comment:
`Section` should have method `renderItems` according to the previous task. You need to use it for rendering the array of cards